packages/odie-client/src/data/use-manage-support-interaction.ts

```diff
@@ -38,15 +38,36 @@ export const useManageSupportInteraction = () => {
 	/**
 	 * Add an event to a support interaction.
 	 */
-	const addEventToInteraction = useMutation( {
+	const addEventToInteraction = useMutation<
+		SupportInteraction,
+		Error,
+		{ interactionId: string; eventData: SupportInteractionEvent }
+	>( {
```

Comment:
In general this is not a good practice, to just export the .mutate

The best way to to export a query (either a mutator or a get) is to export the hole method. Example

```suggestion
	export const useEventInteractionMutation = useMutation<
		SupportInteraction,
		Error,
		{ interactionId: string; eventData: SupportInteractionEvent }
	>( {
```


That way, when using the mutator, we can extracty ANYTHING we'd like from the mutator. Like for example, to know if the mutator isLoading.

When we'd want to use the mutator, we import the mutator like.

```tsx
const {mutate: addEventToInteraction, isLoading } = useEventInteractionMutation();
...
addEventToInteraction( params );

if ( isLoading ) {
   return <MyFancyLoader/>;
}
```

Or even if we need to se the async mutator, it's as easy as:
```tsx
const {mutateAsync: addEventToInteraction } = useEventInteractionMutation();
...
try {
    setLoader( true );
    await addEventToInteraction( params );
} catch (exception e) {
    setErrorMessage("Please try again later");
} finally {
    setLoader( false );
}
```